docs/topics/jvm/java-interop.md

```diff
@@ -137,20 +144,76 @@ Set<@NotNull String> toSet(@NotNull Collection<@NotNull String> elements) { ...
 It leads to the following signature seen in Kotlin:
 
 ```kotlin
-fun toSet(elements: (Mutable)Collection<String>) : (Mutable)Set<String> { ... }
+fun toSet(elements: (Mutable)Collection<String>) : (Mutable)Set<String>
 ```
 
 Note the `@NotNull` annotations on `String` type arguments. Without them, you get platform types in the type arguments:
 
 ```kotlin
-fun toSet(elements: (Mutable)Collection<String!>) : (Mutable)Set<String!> { ... }
+fun toSet(elements: (Mutable)Collection<String!>) : (Mutable)Set<String!>
+```
+
+Kotlin also takes into account nullability annotations on type arguments of base classes and interfaces. For example,
+there are 2 Java classes with the signatures provided below.
+
+```java
+public class Base<T>
+```
+
+```java
+public class Derived extends Base<@Nullable String>
+```
+
+In the Kotlin code, passing the instance of `Derived` where the `Base<String>` assumed produces the warning.
+
+```kotlin
+fun takeBaseOfNotNullStrings(x: Base<String>) {}
+
+fun main() {
+    takeBaseOfNotNullStrings(Derived()) // warning 
+}
+```
+
+#### Type parameters
+
+By default, plain type parameters both in Kotlin and Java have undefined nullability. From Java, you can specify 
+it with nullability annotations. Let's annotate the type parameter of the `Base` class:
+
+```java
+public class Base<@NotNull T>
+```
+
+In Kotlin, inheriting from `Base` expects a non-nullable type argument or type parameter. Thus, the following Kotlin code
+produces warning:
```

Comment:
I'd say "warning by default for now" and that it's going to become an error in the future (in 1.6.0, but I think it's not worth specifying a specific version), because it may seem to a user that the compiler will report warnings permanently.

Reply:
Our docs reflect an actual state for the latest stable version, and we avoid announcing our plans for the future.
There are general reasons:
- This info complicates the text while being irrelevant for some readers.
- Plans can change.
- "for now" raises more questions than it gives answers =)

Aside from that, the planned change (warning > error) will require a doc update anyway, so no doubt that we'll return to this content when it's time.